lockin/src/lib.rs

```diff
@@ -0,0 +1,973 @@
+#![no_std]
+use core::cmp::Ord;
+use core::convert::From;
+use core::ops::SubAssign;
+use core::f32::consts::PI;
+
+extern crate libm;
+
+mod trig;
+use trig::{atan2, cos, sin};
+
+// TODO temporary, pending new DMA code
+pub const SAMPLE_BUFFER_SIZE: usize = 16;
+pub const TSTAMP_BUFFER_SIZE: usize = SAMPLE_BUFFER_SIZE / 2;
+pub const OUTPUT_BUFFER_SIZE: usize = 1;
+
+/// Slow external reference edge timestamps.
+#[derive(Copy, Clone)]
+pub struct TimeStamp {
```

Comment:
This can just be a `t: i32`. The age increment per batch for an old timestamp would be `t -= something`.